File: packages/sdk-ts/examples/agent-indicator.ts
import "dotenv/config";
import { Stagehand } from "../src/index.js";

const { BROWSERBASE_API_KEY } = process.env;
if (!BROWSERBASE_API_KEY) {
  throw new Error("BROWSERBASE_API_KEY is required");
}

const stagehand = new Stagehand({
  apiKey: BROWSERBASE_API_KEY,
  agentIndicator: true,
  browser: {
    type: "browserbase",
    userMetadata: { demo: "agent-indicator" },
  },
  logging: { level: "info" },
});

const sleep = (milliseconds: number) =>
  new Promise<void>((resolve) => globalThis.setTimeout(resolve, milliseconds));

try {
  await stagehand.init();

  const sessionId = stagehand.browser.browserbaseSessionId;
  if (!sessionId) throw new Error("Browserbase did not return a session ID");

  console.log(`View the live browser: https://www.browserbase.com/sessions/${sessionId}`);

  const page = (await stagehand.context.pages())[0] ?? (await stagehand.context.newPage());
  await page.goto("https://example.com", { waitUntil: "load" });

  console.log("The orange halo will remain visible for 60 seconds.");
  await sleep(60_000);
} finally {
  await stagehand.close();
}

File: packages/sdk-python/examples/agent_indicator.py
import asyncio

from stagehand import Stagehand


async def main() -> None:
    stagehand = Stagehand(
        browser="local",
        headless=False,
        agent_indicator=True,
    )

    try:
        await stagehand.init()

        pages = await stagehand.context.pages()
        page = pages[0] if pages else await stagehand.context.new_page()

        print("Navigating: the orange indicator should already be active.")
        await page.goto("https://example.com", wait_until="load")
        await asyncio.sleep(1.5)

        print(f"Reading the page title: {await page.title()}")
        await asyncio.sleep(4)
        print("The indicator should still be active. Closing Stagehand now.")
    finally:
        await stagehand.close()


if __name__ == "__main__":
    asyncio.run(main())
